web/packages/shared/components/TextEditor/TextEditor.jsx

```diff
@@ -104,20 +112,58 @@ class TextEditor extends React.Component {
 
   render() {
     const { bg = 'levels.sunken' } = this.props;
+    const hasButton = this.props.copyButton || this.props.downloadButton;
+
     return (
       <StyledTextEditor bg={bg}>
         <div ref={e => (this.ace_viewer = e)} />
+        {hasButton && (
+          <ButtonSection>
+            {this.props.copyButton && (
+              <EditorButton
+                title="Copy to clipboard"
+                onClick={() => copyToClipboard(this.editor.session.getValue())}
+              >
+                <Copy size="medium" />
+              </EditorButton>
+            )}
+            {this.props.downloadButton && (
+              <EditorButton
+                title="Download"
+                onClick={() =>
+                  downloadObject(
+                    this.props.downloadFileName,
+                    this.editor.session.getValue()
+                  )
+                }
+              >
+                <Download size="medium" />
+              </EditorButton>
+            )}
+          </ButtonSection>
+        )}
       </StyledTextEditor>
     );
   }
 }
 
-export default TextEditor;
-
 function getMode(docType) {
   if (docType === 'json') {
     return 'ace/mode/json';
   }
 
   return 'ace/mode/yaml';
 }
+
+const EditorButton = styled(ButtonSecondary)`
+  padding: ${({ theme }) => theme.space[2]}px;
```

Comment:
I have some issues with hovering the button, I think the ace editor has a hidden scrollbar on the right side that covers the button. 
Probably adding a minimal z-index will help.
https://github.com/gravitational/teleport/assets/20583051/735ffd95-4f68-484a-a11f-8606d2eae28d

Reply:
Good catch, thank you.

It looks like the scroll bar's index is `6`, so I added a `z-index` of 10 to the button section. It fixed the problem, ty.